app/services/importers/npq_manual_validation.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class Importers::NPQManualValidation
+  attr_reader :path_to_csv
+
+  def initialize(path_to_csv:)
+    @path_to_csv = path_to_csv
+  end
+
+  def call
+    check_headers
+
+    rows.each do |row|
+      data = NPQValidationData.find_by(id: row["application_ecf_id"])
+
+      puts "no NPQValidationData found for #{row['application_ecf_id']}" if data.nil?
+      next if data.nil?
+
+      puts "updating trn for NPQValidationData: #{row['application_ecf_id']} with trn: #{row['validated_trn']}"
+
+      data.update!(teacher_reference_number: row["validated_trn"], teacher_reference_number_verified: true)
+
+      NPQ::CreateOrUpdateProfile.new(npq_validation_data: data).call
```

Comment:
I can see you getting errors here if there's already a teacher profile with that TRN

Reply:
This should probably be wrapped in a transaction

Reply:
Sure can wrap in transaction however don't follow where the error would be. Looking at the code multiple `TeacherProfile`s with the same TRN are permitted. `NPQ::CreateOrUpdateProfile` only creates or updates the TRN on the singleton `TeacherProfile`

Reply:
it's in a transaction now